test/test_jit.py

```diff
@@ -6499,6 +6499,78 @@ def test_index_put(target, indices, rhs):
 
         self.assertExpectedGraph(test_index_put.graph)
 
+    def test_tuple_indexing(self):
+        @torch.jit.script
+        def tuple_index(a):
+            if a:
+                b = (1, 2)
+            else:
+                b = (0, 2)
+            return b[-2], b[1]
+
+        self.assertExpectedGraph(tuple_index.graph)
+        self.assertEqual(tuple_index(torch.tensor([1])), (1, 2))
+        self.run_pass('lower_all_tuples', tuple_index.graph)
+        self.assertEqual(tuple_index(torch.tensor([1])), (1, 2))
+
+        with self.assertRaisesRegex(RuntimeError, "Indexing on tuples only supported with integer constants:"):
+            @torch.jit.script
```

Comment:
Every one of these tests needs to actually invoke the functions they write using `self.checkScript` (not sure that is the right name, but the function that compares script against a normal python function.